app/components/header.tsx

```diff
@@ -94,8 +99,25 @@ export default function Header() {
 
       <Flex justifyContent="flex-end" alignItems="center" color="white" gap="5" width="100%">
         <Hide below="lg">
-          <Text id="header-user">{user?.username}</Text>
+          <Text id="header-user">
+            {user.username === originalUser.username
+              ? user?.username
+              : `Mocking: ${user?.username}`}
+          </Text>
         </Hide>
+        {user.username !== originalUser.username && (
+          <Tooltip label="Revert to original user">
```

Comment:
I wonder if a better place for this is in the sub-menu where we have "Sign-out".  So you can "Stop Impersonating" or "Sign Out" (2 options).  Then it wouldn't be so cluttered in the header.

Another idea: change the icon for the User Avatar to the Impersonate masks, to indicate that's what's happening?